pennylane/ftqc/qubit_graph.py

```diff
@@ -0,0 +1,401 @@
+# Copyright 2025 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""This module contains the data structures to represent the qubit memory model in the FTQC/MBQC
+framework, as well as the API to access and manipulate these data structures.
+"""
+
+import warnings
+from typing import Union
+
+import networkx as nx
+
+
+class QubitGraph:
+    """A class to represent nested graphs of qubits in the FTQC/MBQC framework.
+
+    A QubitGraph is a qubit that contains a graph of underlying qubits, where each underlying qubit
+    is itself a QubitGraph. This representation allows for nesting of lower-level qubits with
+    arbitrary depth to allow easy insertion of arbitrarily many levels of abstractions between
+    logical qubits and physical qubits.
+
+    Args:
+        graph (graph-like, optional): The graph to use as the QubitGraph's underlying qubits.
+            Inputting None (the default), leaves the underlying qubit graph in an uninitialized
+            state, in which case one of the graph-initialization methods may be used to define the
+            structure of the underlying qubit graph. QubitGraph expects an undirected graph as
+            input, specifically an instance of the ``networkx.Graph`` class, although other networkx
+            graphs and graph-like types are also permitted. An object is considered "graph-like" if
+            it has both a 'nodes' and an 'edges' attribute.
+
+    TODO:
+
+        * How to represent operations on qubits?
+            * We should be able to broadcast operations to underlying qubits, assuming operations
+              are transversal.
+            * Recall that a _transversal operation_ is defined as a logical operator that is formed
+              by applying the individual physical operators to each qubit in a QEC code block.
+        * Implement tensor-like indexing and slicing.
+        * Improve string representation of QubitGraph objects.
+    """
+
+    def __init__(self, graph: nx.Graph = None):
```

Comment:
```suggestion
    def __init__(self, graph: Optional[nx.Graph] = None):
```
If we expect this to also be `None` we can declare the type as [optional](https://docs.python.org/3/library/typing.html#typing.Optional)